packages/backend/src/tokens/request.ts

```diff
@@ -125,27 +125,26 @@ export async function authenticateRequest(
     try {
       verifyResult = await verifyToken(sessionToken, authenticateContext);
     } catch (err) {
-      if (err instanceof TokenVerificationError) {
+      if (
+        err instanceof TokenVerificationError &&
+        instanceType === 'development' &&
+        (err.reason === TokenVerificationErrorReason.TokenExpired ||
+          err.reason === TokenVerificationErrorReason.TokenNotActiveYet)
+      ) {
```

Comment:
Previously, we weren't re-throwing the error for production, which is incorrect. In the scenario where verification fails on the token from the handshake, production should treat this as a fatal error.